monix-eval/shared/src/test/scala/monix/eval/TaskConversionsSuite.scala

```diff
@@ -378,6 +379,51 @@ object TaskConversionsSuite extends BaseTestSuite {
     assertEquals(f2.value, Some(Success(99)))
   }
 
+  test("Task.fromReactivePublisher protects against user error") { implicit s =>
+
+    val dummy = DummyException("dummy")
+
+    val pub = new Publisher[Int] {
+      def subscribe(s: Subscriber[_ >: Int]): Unit = {
+        s.onSubscribe(new Subscription {
+          def request(n: Long): Unit = throw dummy
+          def cancel(): Unit = throw dummy
+        })
+      }
+    }
+
+    assertEquals(Task.fromReactivePublisher(pub).runToFuture.value, Some(Failure(dummy)))
+  }
+
+  test("Task.fromReactivePublisher yields expected input") { implicit s =>
+
+    val pub = new Publisher[Int] {
+      def subscribe(s: Subscriber[_ >: Int]): Unit = {
+        s.onSubscribe(new Subscription {
+          var isActive = true
+          def request(n: Long): Unit = {
+            if (n > 0 && isActive) {
+              isActive = false
+              s.onNext(1)
+              s.onComplete()
```

Comment:
I think this call is irrelevant here, what do you think?

Reply:
I think it's fine, it doesn't change the test but it is something that would happen in real scenario so we might as well keep it

Reply:
Sounds reasonable 👍